prow/caching.md

```diff
@@ -0,0 +1,50 @@
+# Caching dependencies in Prow
+
+Caching dependencies in prow works only when the [Pod Utilities][0] are enabled. A
+valid GCS Bucket has to be configured via `gcs_configuration` either in the Prow
+jobs `decoration_config` or in [Planks][1] `default_decoration_configs`.
+
+[0]: /prow/pod-utilities.md
+[1]: /prow/cmd/plank/README.md
+
+Saving and restoring the cache will be done in the `clonerefs` and `sidecar` pod
+utility containers, where:
+
+- `clonerefs` tries to restore the cache after the clone has been done. Multiple
+  `keys` can be specified to reference files inside the repository. Those files
+  will be hashed and serve as unique identifier for the cache.
+
+- `sidecar` tries to save the cache after the main test container has been
+  finished successfully. The specified `keys` will be used here as well to
+  calculate the target directory.
+
+We store the cache in Prows global root directory of the Bucket (`/prow/cache`),
+because the cache should be available between multiple Prow runs and even
+between different Prow jobs.
+
+A single cache entry can be defined in the `decoration_config` via:
+
+```yaml
+decoration_config:
+  caches:
+    - name: my-cache
+      path: /gocache
+      keys:
+        - go.mod
+        - go.sum
+```
+
+Please ensure that the `name` has to be unique because it will be used as
+reference for the internally used volumes and mounts. Another point worth
+mentioning is that the `path` has to be absolute to be mount-able.
+
+The `keys` contain repository-relative paths, where a SHA256 is calculated for
+each of the specified files. The complete cache path on the remote bucket will
+then be assembled like this:
+
+```
+/prow/cache/{{ name }}-{{ sha256 keys[0] }}-{{ sha256 keys[1] }}…
```

Comment:
I think it would make sense to upload this is a `tar.gz` rather than individually. My local gocache which I cleared yesterday has for example more then 30k files already, uploading all of this individually is very inefficient

Reply:
I agree that makes sense.